system/osx_defaults.py

```diff
@@ -160,6 +172,15 @@ def _convert_type(self, type, value):
             if not isinstance(value, list):
                 raise OSXDefaultsException("Invalid value. Expected value to be an array")
             return value
+        elif type in ["dict", "dictionary"]:
+            if isinstance(value, dict):
+                return [value]
+            elif isinstance(value, list):
+                return value
+            elif isinstance(value, str):
```

Comment:
I could use some eyes and knowledge on this. In 2.0, value came in from a dictionary in role defaults value as a dict type. Now in 2.1 it's coming through as a str. Eval works here, but I don't like it.